### ISSUE TITLE
selection/search: if select2 is created in an invisible element then it'll have placeholder with width=100px

### ISSUE THREAD

**Issue by evil-shrike** [select2/select2#4513]
"placeholder with width:100px" means that most likely it'll be cut.
It happens only for multi-select. Because of its `Search.resizeSearch`:

selection/search

```
  Search.prototype.resizeSearch = function () {
    this.$search.css('width', '25px');

    var width = '';

    if (this.$search.attr('placeholder') !== '') {
      width = this.$selection.find('.select2-selection__rendered').innerWidth();
    } else {
      var minimumWidth = this.$search.val().length + 1;

      width = (minimumWidth * 0.75) + 'em';
    }

    this.$search.css('width', width);
  };
```

`$(.select2-selection__rendered').innerWidth()` returns "100" if element is invisible.
Actually I'm rendering select2 in Bootstrap dialog which is being shown with animation so at the time of resizeSearch select is still hidden.

```
> this.$selection.find('.select2-selection__rendered').css("width")
"100%"
> this.$selection.find('.select2-selection__rendered').innerWidth()
100
```

jQuery docs says "The value reported by .innerWidth() is not guaranteed to be accurate when the element or its parent is hidden."
jQuery: v1.12.4

**Comment by jmiguel-localvox** [comment]
I can confirm this behavior. 

A work around (if you can identify when that field is going to show up) is to call `$target.data('select2').selection.resizeSearch()`.

**Comment by Ceasar** [comment]
This issue is actually slightly more problematic: because the width it set when the Select2 is invoked and not dynamically, if you resize the viewport the element may be a different width than what it would be if the viewport was originally the new size.

**Comment by alexweissman** [comment]
There is a fix for this in #4898 under discussion.  Please see that PR for my thoughts.

**Comment by alexweissman** [comment]
Also, I am going to close this issue because it is a duplicate of a _very_ old issue, that is still under discussion in #291.
